sale_sourced_by_line/model/sale.py

```diff
@@ -1,66 +1,33 @@
 # -*- coding: utf-8 -*-
-#
-#
-#    Author: Guewen Baconnier, Yannick Vaucher
-#    Copyright 2013-2015 Camptocamp SA
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as
-#    published by the Free Software Foundation, either version 3 of the
-#    License, or (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-#
+# Copyright 2013-2014 Camptocamp SA - Guewen Baconnier
+# © 2016 Eficent Business and IT Consulting Services S.L.
+# © 2016 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
-from openerp import models, api, fields
-from openerp.osv import orm
+
+from openerp import api, fields, models
 
 
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
-    def _prepare_order_line_procurement(self, cr, uid, order, line,
-                                        group_id=False, context=None):
-        values = super(SaleOrder, self)._prepare_order_line_procurement(
-            cr, uid, order, line, group_id=group_id, context=context)
-        if line.warehouse_id:
-            values['warehouse_id'] = line.warehouse_id.id
-        return values
-
     @api.model
     def _prepare_procurement_group_by_line(self, line):
         vals = super(SaleOrder, self)._prepare_procurement_group_by_line(line)
         # for compatibility with sale_quotation_sourcing
-        if line._get_procurement_group_key()[0] == 8:
+        if line._get_procurement_group_key()[0] == 10:
             if line.warehouse_id:
                 vals['name'] += '/' + line.warehouse_id.name
         return vals
 
-    SO_STATES = {
-        'cancel': [('readonly', True)],
-        'progress': [('readonly', True)],
-        'manual': [('readonly', True)],
-        'shipping_except': [('readonly', True)],
-        'invoice_except': [('readonly', True)],
-        'done': [('readonly', True)],
-    }
```

Comment:
Any comment or reason about removing this?

Was added in https://github.com/OCA/sale-workflow/commit/40c864e498ad2ebbb6bc4030718c61853cd58bad
which in fact doesn't give more clue why it is here.

To me, it seems reasonable to have that field only editable in draft or open states as once `_prepare_order_line_procurement` has been called there is no reason that I can think of to change this value.

In fact to be consistent with other fields I think it should be:

    warehouse_id = fields.Many2one(
        [...]
        readonly=True,
        states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}